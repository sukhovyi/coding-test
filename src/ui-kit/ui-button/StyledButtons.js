import { styled } from '@egoist/vue-emotion';
import Theme from '../Theme';

const Button = styled('button')`
  font-size: 14px;
  font-weight: bold;
  border: none;
  padding: 10px;
  border-radius: 5px;
`;

export const PrimaryButton = styled(Button)`
  background-color: ${Theme.colors.primary};
  color: #FFFFFF;
`;

export const DangerButton = styled(Button)`
  background-color: ${Theme.colors.danger};
  color: #FFFFFF;
`;

export const SuccessButton = styled(Button)`
  background-color: ${Theme.colors.success};
  color: #FFFFFF;
`;

export const SecondaryButton = styled(Button)`
  background-color: ${Theme.colors.secondary};
`;

export const TertiaryButton = styled(Button)`
  background-color: ${Theme.colors.tertiary};
`;
