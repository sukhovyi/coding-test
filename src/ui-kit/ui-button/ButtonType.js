import { PrimaryButton, SecondaryButton, DangerButton, SuccessButton, TertiaryButton } from './StyledButtons';

const ButtonType = Object.freeze({
  PRIMARY: 'primary',
  DANGER: 'danger',
  SUCCESS: 'success',
  SECONDARY: 'secondary',
  TERTIARY: 'tertiary',

  getButtons() {
    return {
      PrimaryButton,
      SecondaryButton,
      DangerButton,
      SuccessButton,
      TertiaryButton,
    };
  },

  getButtonByType(type) {
    const button = {
      [ButtonType.PRIMARY]: PrimaryButton,
      [ButtonType.DANGER]: DangerButton,
      [ButtonType.SUCCESS]: SuccessButton,
      [ButtonType.SECONDARY]: SecondaryButton,
      [ButtonType.TERTIARY]: TertiaryButton,
    }[type];

    if (!button) {
      throw new Error(`Button with type ${type} does not exist!`);
    }

    return button;
  },
});

export default ButtonType;
