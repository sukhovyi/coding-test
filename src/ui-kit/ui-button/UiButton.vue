<template>
  <component :is="buttonComponent" @click="onClick">
    <slot/>
  </component>
</template>

<script>
import ButtonType from './ButtonType';

export default {
  name: 'UiButton',
  components: {
    ...ButtonType.getButtons(),
  },
  props: {
    type: {
      type: String,
      default: ButtonType.PRIMARY,
    },
  },
  computed: {
    buttonComponent() {
      return ButtonType.getButtonByType(this.type);
    },
  },
  methods: {
    onClick() {
      this.$emit('click');
    },
  },
};
</script>
