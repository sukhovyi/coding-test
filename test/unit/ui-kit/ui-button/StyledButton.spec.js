import { PrimaryButton, SecondaryButton, DangerButton, SuccessButton, TertiaryButton } from '../../../../src/ui-kit/ui-button/StyledButtons';

describe('ui-kit/ui-button/StyledButton', () => {
  it('PrimaryButton should have correct styling', () => {
    expect(PrimaryButton).toMatchSnapshot();
  });

  it('SecondaryButton should have correct styling', () => {
    expect(SecondaryButton).toMatchSnapshot();
  });

  it('DangerButton should have correct styling', () => {
    expect(DangerButton).toMatchSnapshot();
  });

  it('SuccessButton should have correct styling', () => {
    expect(SuccessButton).toMatchSnapshot();
  });

  it('TertiaryButton should have correct styling', () => {
    expect(TertiaryButton).toMatchSnapshot();
  });
});
