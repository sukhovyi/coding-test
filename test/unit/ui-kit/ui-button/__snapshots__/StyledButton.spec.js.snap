// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ui-kit/ui-button/StyledButton DangerButton should have correct styling 1`] = `
Object {
  "__emotion_base": "button",
  "__emotion_real": [Circular],
  "__emotion_styles": Array [
    "
  font-size: 14px;
  font-weight: bold;
  border: none;
  padding: 10px;
  border-radius: 5px;
  cursor: pointer;
",
    "
  background-color: ",
    "#CC2E2E",
    ";
  color: #FFFFFF;
",
  ],
  "functional": true,
  "inject": Object {
    "theme": Object {
      "default": undefined,
    },
  },
  "name": "Styledbutton",
  "render": [Function],
  "withComponent": [Function],
}
`;

exports[`ui-kit/ui-button/StyledButton PrimaryButton should have correct styling 1`] = `
Object {
  "__emotion_base": "button",
  "__emotion_real": [Circular],
  "__emotion_styles": Array [
    "
  font-size: 14px;
  font-weight: bold;
  border: none;
  padding: 10px;
  border-radius: 5px;
  cursor: pointer;
",
    "
  background-color: ",
    "#0063D9",
    ";
  color: #FFFFFF;
",
  ],
  "functional": true,
  "inject": Object {
    "theme": Object {
      "default": undefined,
    },
  },
  "name": "Styledbutton",
  "render": [Function],
  "withComponent": [Function],
}
`;

exports[`ui-kit/ui-button/StyledButton SecondaryButton should have correct styling 1`] = `
Object {
  "__emotion_base": "button",
  "__emotion_real": [Circular],
  "__emotion_styles": Array [
    "
  font-size: 14px;
  font-weight: bold;
  border: none;
  padding: 10px;
  border-radius: 5px;
  cursor: pointer;
",
    "
  background-color: ",
    "#E7E7E7",
    ";
",
  ],
  "functional": true,
  "inject": Object {
    "theme": Object {
      "default": undefined,
    },
  },
  "name": "Styledbutton",
  "render": [Function],
  "withComponent": [Function],
}
`;

exports[`ui-kit/ui-button/StyledButton SuccessButton should have correct styling 1`] = `
Object {
  "__emotion_base": "button",
  "__emotion_real": [Circular],
  "__emotion_styles": Array [
    "
  font-size: 14px;
  font-weight: bold;
  border: none;
  padding: 10px;
  border-radius: 5px;
  cursor: pointer;
",
    "
  background-color: ",
    "#3FBA69",
    ";
  color: #FFFFFF;
",
  ],
  "functional": true,
  "inject": Object {
    "theme": Object {
      "default": undefined,
    },
  },
  "name": "Styledbutton",
  "render": [Function],
  "withComponent": [Function],
}
`;

exports[`ui-kit/ui-button/StyledButton TertiaryButton should have correct styling 1`] = `
Object {
  "__emotion_base": "button",
  "__emotion_real": [Circular],
  "__emotion_styles": Array [
    "
  font-size: 14px;
  font-weight: bold;
  border: none;
  padding: 10px;
  border-radius: 5px;
  cursor: pointer;
",
    "
  background-color: ",
    "#AAAAAA",
    ";
",
  ],
  "functional": true,
  "inject": Object {
    "theme": Object {
      "default": undefined,
    },
  },
  "name": "Styledbutton",
  "render": [Function],
  "withComponent": [Function],
}
`;
